src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -300,7 +303,7 @@ public Task<Document> AddEventAsync(Solution solution, INamedTypeSymbol destinat
             return GetEditAsync(
                 solution,
                 destination,
-                (t, opts, ai) => AddEvent(t, @event, opts, ai),
+                (t, opts, ai, _) => AddEvent(t, @event, opts, ai),
```

Comment:
Only types and namespaces deal with CancellationToken. CancellationTokens are always checked in GetMembers().

Note that with this approach, cancellation for MAS of Microsoft.CodeAnalysis.Shared.Extensions.ObjectExtensions (TypeSwitch) is still imperceptibly fast.